modules/messages/client/components/Inbox.component.js

```diff
@@ -0,0 +1,134 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { useTranslation } from 'react-i18next';
+
+import * as api from '../api/messages.api';
+import Avatar from '@/modules/users/client/components/Avatar.component';
+import Activate from '@/modules/users/client/components/Activate';
+import { $broadcast, eventTrack } from '@/modules/core/client/services/angular-compat';
+import TimeAgo from '@/modules/core/client/components/TimeAgo';
+
+const userType = PropTypes.shape({
+  _id: PropTypes.string.isRequired,
+  username: PropTypes.string.isRequired,
+  displayName: PropTypes.string.isRequired,
+});
+
+export default function Inbox({ user }) {
+  if (!user.public) {
+    return (
+      <section className="container-spacer">
+        <Activate/>
+      </section>
+    );
+  }
+  const { t } = useTranslation('messages');
+
+  const [nextParams, setNextParams] = useState(null);
+  const [isFetching, setIsFetching] = useState(false);
+  const [threads, setThreads] = useState([]);
```

Comment:
Not opposed to this, but is the idea we never use `this.state` anymore now that we have hooks? 😁 

I'm still kinda uneducated about React Hooks so seriously curious! Not asking to change anything here either. :-)

Reply:
Yup, no `this.state` anymore, :fire: it all (well, perhaps there are things that only work with class components, and `this.state`, but I didn't come across anything).

Actually, later on the idea would be to use (or at least consider using) redux for some of the state management.

When you say _not opposed to this_, does that mean you have some caution/hesitation?

Reply:
Just haven't seen the Hooks used so much yet for everything stateful in a component (rather one off things), but I haven't seen hooks in action much yet at all. 

It's all good, thanks for elaborating!